File: 07/julia/sol1.jl
# Load input
using DelimitedFiles
inp = readdlm("../input", ',', Int) .+ 1

# Tally crab locations
m = maximum(inp)
crabs = zeros(Int, m)
view(crabs, inp) .+= 1

# Calculate distance travelled for each candidate location
distances = abs.(repeat(1:m, inner=(1,m))' .- (1:m))
travelled = distances * crabs

# Print the best
println(minimum(travelled))

File: 07/octave/sol1.m
# Load crabs
inp = str2num(fileread("../input")) + 1;
m = max(inp);
crabs = histc(inp, 1:m);

# Calculate distances
distances = abs(repmat(1:m, [m,1])' - (1:m));
travelled = distances * crabs';

# Print minimum distance
min(travelled)
